lti_consumer/signals.py

```diff
@@ -0,0 +1,27 @@
+"""
+LTI Consumer related Signal handlers
+"""
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from lti_consumer.models import LtiAgsScore
+from lti_consumer.plugin import compat
+
+
+@receiver(post_save, sender=LtiAgsScore, dispatch_uid='publish_grade_on_score_update')
+def publish_grade_on_score_update(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Publish grade to xblock whenever score saved/updated and its grading_progress is set to FullyGraded.
+    """
+    if instance.grading_progress == LtiAgsScore.FULLY_GRADED:
+        block = compat.load_block_as_anonymous_user(instance.line_item.resource_link_id)
+        if not block.is_past_due():
```

Comment:
@shimulch There's a `accept_grades_past_due` setting in the block that needs to be respected here.
```suggestion
        if not block.is_past_due() or block.accept_grades_past_due:
```

Reply:
https://github.com/open-craft/xblock-lti-consumer/blob/bbd55a49d31849e6578f4130ad60f14bf92ab9ef/lti_consumer/lti_xblock.py#L466

Reply:
@giovannicimolin added that on the condition and added test for that.